harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -636,3 +666,77 @@ def test_checkpoint_loading(ckpt: str, istrial: bool):
         assert isinstance(trial, pytorch.PyTorchTrial), type(trial)
     else:
         assert isinstance(trial, torch.nn.Module), type(trial)
+
+
+def make_amp_workloads(
+    trial_class,
+    assert_output_float16=False,
+) -> workload.Stream:
+    trainer = utils.TrainAndValidate()
+    yield from trainer.send(steps=20, validation_freq=1, scheduling_unit=1)
+    training_metrics, _ = trainer.result()
```

Comment:
This is definitely a nit.

But it is possible to write this in a way that the assertions happen in the body of the test instead of in the workloads function.  It's not materially different, just slightly cleaner I think.  You would need to use the `nonlocal` to manually bind an external variable into the function definition so you can write to it inside the bound function and read from it after the bound function is complete.  It would look like this:

```
def test_amp():
    
    training_metrics = None

    def make_workloads() -> workload.Stream:
        trainer = utils.TrainAndValidate()
        yield from trainer.send(steps=20, scheduling_unit=1)
        
        nonlocal training_metrics
        training_metrics, _ = trainer.result()

    controller = ...
    controller.run()

    # by now training_metrics will be defined
    assert training_metrics

Reply:
I would love to do this, but we should do it for all `PyTorchTrial` tests. If it's not too much of an abuse of the task scope, I'd be happy to do it now, but I don't want to make that call on my own. We can always make a new task, etc., but this would be a pretty quick change.

Reply:
no need to grow the scope of this change beyond the code you are writing.

But the question of asserts inside the workloads function is distinct from having `make_amp_workloads` as a standalone function (defined outside the scope of `test_amp()`).  This _is_ different than all the other tests in `test_pytorch_trial.py`.

Reply:
How about we revisit this possibility when I get to [DET-1597](https://determinedai.atlassian.net/browse/DET-1597)?

Reply:
This was a nit-pick, and not a big deal.  Land it how you like...

> How about we revisit this possibility

... but let's not land something we plan on changing right away.  I'm fine leaving it as-written indefinitely.  If you are ok with that, let's plan on that.